Compilation without OpenMP fails in iop/denoiseprofile.c

**Describe the bug/issue**
Compilation fails with OpenMP functions disabled.

**To Reproduce**
Compile on Windows 10 (20H2) in an up-to-date MSYS2 system in a clean build directory.
```
cmake -G "Ninja" -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=./release -DUSE_OPENMP=False ..
cmake --build .
```
This halts here:
```
[447/683] Building C object lib/darktable/plugins/CMakeFiles/denoiseprofile.dir/introspection_denoiseprofile.c.obj
FAILED: lib/darktable/plugins/CMakeFiles/denoiseprofile.dir/introspection_denoiseprofile.c.obj
C:\msys64\mingw64\bin\cc.exe -DAVIF_DLL -DGDK_DISABLE_DEPRECATED -DGDK_VERSION_MIN_REQUIRED=GDK_VERSION_3_22 -DGLIB_VERSION_MAX_ALLOWED=GLIB_VERSION_MIN_REQUIRED -DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_56 -DGTK_DISABLE_DEPRECATED -DGTK_DISABLE_SINGLE_INCLUDES -DHAVE_BUILTIN_CPU_SUPPORTS -DHAVE_CONFIG_H -DHAVE_GAME -DHAVE_GMIC -DHAVE_GPHOTO2 -DHAVE_GPHOTO_25_OR_NEWER -DHAVE_GRAPHICSMAGICK -DHAVE_HTTP_SERVER -DHAVE_ICU -DHAVE_ISO_CODES -DHAVE_LENSFUN -DHAVE_LIBAVIF=1 -DHAVE_LIBSECRET -DHAVE_MAP -DHAVE_OPENCL -DHAVE_OPENEXR -DHAVE_OPENJPEG -DHAVE_OSMGPSMAP_110_OR_NEWER -DHAVE_SQLITE_324_OR_NEWER -DSQLITE_CORE -DSQLITE_ENABLE_ICU -D_POSIX_THREAD_SAFE_FUNCTIONS -D__GDK_KEYSYMS_COMPAT_H__ -D__USE_MINGW_ANSI_STDIO=1 -Ddenoiseprofile_EXPORTS -Ilib/darktable/plugins -I../src/iop -I../src -Ibin -isystem ../src/external -isystem ../src/external/OpenCL -isystem C:/msys64/mingw64/include/glib-2.0 -isystem C:/msys64/mingw64/lib/glib-2.0/include -isystem C:/msys64/mingw64/include/gtk-3.0 -isystem C:/msys64/mingw64/include/pango-1.0 -isystem C:/msys64/mingw64/include/harfbuzz -isystem C:/msys64/mingw64/include/freetype2 -isystem C:/msys64/mingw64/include/libpng16 -isystem C:/msys64/mingw64/include/fribidi -isystem C:/msys64/mingw64/include/cairo -isystem C:/msys64/mingw64/include/lzo -isystem C:/msys64/mingw64/include/pixman-1 -isystem C:/msys64/mingw64/include/gdk-pixbuf-2.0 -isystem C:/msys64/mingw64/include/atk-1.0 -isystem C:/msys64/mingw64/include/libxml2 -isystem C:/msys64/mingw64/include/libsoup-2.4 -isystem C:/msys64/mingw64/include/OpenEXR -isystem C:/msys64/mingw64/include/lensfun -isystem C:/msys64/mingw64/include/librsvg-2.0 -isystem C:/msys64/mingw64/include/json-glib-1.0 -isystem C:/msys64/mingw64/include/openjpeg-2.3 -isystem C:/msys64/mingw64/include/libsecret-1 -isystem C:/msys64/mingw64/include/GraphicsMagick -isystem C:/msys64/mingw64/include/pugixml-1.9 -isystem C:/msys64/mingw64/include/osmgpsmap-1.0 -Wall -Wno-format -Wshadow -Wtype-limits -Wvla -Wold-style-declaration -Wno-unknown-pragmas -Wno-error=varargs -Wno-format-truncation -Wno-error=address-of-packed-member -std=c99 -march=native  -g -O3 -DNDEBUG -O3 -ffast-math -fno-finite-math-only -fexpensive-optimizations -fvisibility=hidden   -DUNICODE -D_UNICODE -Werror -Wfatal-errors -include common/module_api.h -include iop/iop_api.h -MD -MT lib/darktable/plugins/CMakeFiles/denoiseprofile.dir/introspection_denoiseprofile.c.obj -MF lib\darktable\plugins\CMakeFiles\denoiseprofile.dir\introspection_denoiseprofile.c.obj.d -o lib/darktable/plugins/CMakeFiles/denoiseprofile.dir/introspection_denoiseprofile.c.obj -c lib/darktable/plugins/introspection_denoiseprofile.c
In file included from lib/darktable/plugins/introspection_denoiseprofile.c:110:
C:/msys64/home/Roel/dt-dev/src/iop/denoiseprofile.c: In function 'process_wavelets_cl.constprop.isra':
C:/msys64/home/Roel/dt-dev/src/iop/denoiseprofile.c:2202:20: error: iteration 2 invokes undefined behavior [-Werror=aggressive-loop-optimizations]
 2202 |       toRGB[3*k+c] = toRGB_tmp[k][c] * d->strength * scale;
      |       ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated due to -Wfatal-errors.
cc1.exe: all warnings being treated as errors
```

This is tested with 33deb6bb1f49f21d78f4adae4a957cc0f29b58c9

--- a/src/iop/denoiseprofile.c
+++ b/src/iop/denoiseprofile.c
@@ -2197,7 +2197,8 @@ static int process_wavelets_cl(struct dt_iop_module_t *self, dt_dev_pixelpipe_io
   float toY0U0V0[9]; //TODO: change OpenCL kernels to use 3x4 matrices
   float toRGB[9] ;
   for(size_t k = 0; k < 3; k++)
-    for_each_channel(c)
+    for(size_t c = 0; c < 3; c++)
+    //(we can't use for_each_channel here because it can iterate over four elements)
     {
       toRGB[3*k+c] = toRGB_tmp[k][c] * d->strength * scale;
       toY0U0V0[3*k+c] = toY0U0V0_tmp[k][c] / (d->strength * scale);

